hack/local-up-cluster.sh

```diff
@@ -266,17 +267,15 @@ function test_apiserver_off {
     # For the common local scenario, fail fast if server is already running.
     # this can happen if you run local-up-cluster.sh twice and kill etcd in between.
     if [[ "${API_PORT}" -gt "0" ]]; then
-        curl --silent -g ${API_HOST}:${API_PORT}
-        if [ ! $? -eq 0 ]; then
+        if ! curl --silent -g "${API_HOST}:${API_PORT}" ; then
             echo "API SERVER insecure port is free, proceeding..."
         else
             echo "ERROR starting API SERVER, exiting. Some process on ${API_HOST} is serving already on ${API_PORT}"
             exit 1
         fi
     fi
 
-    curl --silent -k -g ${API_HOST}:${API_SECURE_PORT}
-    if [ ! $? -eq 0 ]; then
+    if ! curl --silent -k -g "${API_HOST}:${API_SECURE_PORT}" ; then
```

Comment:
In ./hack/local-up-cluster.sh line 279:
    if [ ! $? -eq 0 ]; then
           ^-- SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.